pkg/tuf/client.go

```diff
@@ -270,33 +277,39 @@ func initializeTUF(mirror string, root []byte, embedded fs.FS, forceUpdate bool)
 		if root == nil {
 			root, err = getRoot(trustedMeta, t.embedded)
 			if err != nil {
-				singletonTUFErr = fmt.Errorf("getting trusted root: %w", err)
+				initErr = fmt.Errorf("getting trusted root: %w", err)
 				return
 			}
 		}
 
 		if err := t.client.Init(root); err != nil {
-			singletonTUFErr = fmt.Errorf("unable to initialize client, local cache may be corrupt: %w", err)
+			initErr = fmt.Errorf("unable to initialize client, local cache may be corrupt: %w", err)
 			return
 		}
 
-		// We may already have an up-to-date local store! Check to see if it needs to be updated.
-		trustedTimestamp, ok := trustedMeta["timestamp.json"]
-		if ok && !isExpiredTimestamp(trustedTimestamp) && !forceUpdate {
-			// We're golden so stash the TUF object for later use
-			singletonTUF = t
-			return
-		}
+		singletonTUF = t
+	})
+	if initErr != nil {
+		return nil, initErr
+	}
 
-		// Update if local is not populated or out of date.
-		if err := t.updateMetadataAndDownloadTargets(); err != nil {
-			singletonTUFErr = fmt.Errorf("updating local metadata and targets: %w", err)
-			return
-		}
+	trustedMeta, err := singletonTUF.local.GetMeta()
```

Comment:
On a subsequent call to this (if the first call failed for some reason), `singletonTUF` might still be nil and therefore panic here.

Reply:
Thanks, this should be fixed now, using `singletonTUFErr` like what was in the code before